.github/workflows/release.yaml

```diff
@@ -0,0 +1,69 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to be released'
+        required: true
+
+permissions:
+  contents: write
+  packages: write
+
+jobs:
+  check:
+    name: preliminary-checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Ensure version is properly set
+        run: |
+          test "$(npm -s run get-version)" = "${{ inputs.tag }}"
```

Comment:
Facing a bit of a dilema over this one, so on one hand we can make sure we set the version right, manually each time (like we already do today). Or on the other hand, we could add a command like this before publishing the packages:

```shell
yarn version --new-version ${{ inputs.tag }} --no-git-tag-version
```

Which will set the right `version` in the `package.json`, and we can leave some kind of placeholder there that we never touch.

If anyone has any opinion on this one, would love to hear it.

Reply:
I'd vote for the first option. I think requiring the version to be released as input can in some cases avoid surprises.

I am thinking about the case in which one forgets to bump the version in `package.json` and tries to release.

Reply:
Personally I'd prefer that we automate the bump of `package.json`. It has been a source of a lot of overhead and conflict resolution time spent (= extra developer time and mental overhead that could have been useful to other things). Ideally it is done in a way that doesn't cause a conflict on the forward-port branches, i.e. the bump doesn't get applied on any of the forward-ported branches automatically.

Reply:
Actually I thought about something else, but as François requested me to open a TAD, let me write that doc, and I'll try to elaborate a couple of options we can chose to tackle this.

Reply:
I'd say the action should bump and commit the bump on release so we don't do it manually.

And for the forward port we should avoid this I think for linting. 

We should only produce new versions of linters on the upgrade of nodejs & typescript, so rarely.

And when upgrading components, we should upgrade and re lint on every dev branch to minimize conflicts between versions

Reply:
Created the TAD and added a mention of this topic under the [Automation workflow section](https://scality.atlassian.net/wiki/spaces/OS/pages/2943090715/Publishing+npm+packages+to+npm+registry#Automation-workflow)

Reply:
While generally I think we should try to address that topic, here are many implications here : what does it mean to trigger a release, multiple interractions with bert-e, need to actually have a version in package.json (esp. when different version are used, transitively, ...), release process, ...

So i think the better way is to do it incrementally: first we handle the use of registry (keeping the current method of manual commit bump), and then we can have a separate discussion on how to move forward to simplify version bump.